### ISSUE TITLE
Remote auto save not allowed for scheduled posts.

### ISSUE THREAD

**Issue by develric** [wordpress-mobile/WordPress-FluxC-Android#1297]
This is related to the major work done for #1247.

Current implementation of remote auto save (branch master-remote-auto-save 48d95ebeaa419601448b39404db732b4ca0663b8) as of today does not allows to **remote auto save scheduled posts**:
https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/48d95ebeaa419601448b39404db732b4ca0663b8/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java#L247

This came out working on the PR [WP-Android#10085](https://github.com/wordpress-mobile/WordPress-Android/pull/10085) (please refer to there for additional background). 

We decided to manage the scheduled posts in the same way as the published posts, that is using the remote auto save to create an unpublished revision to be shown as a remote preview to the user without affecting the published (or scheduled) version.

For the above feature to be implemented we need the remote auto save of scheduled posts in FluxC.

**NOTE:** next step after post is to enable the remote preview feature also for pages, so the same principle as above applies (to be kept in mind while working on posts).

cc @maxme and @malinajirka

**Comment by malinajirka** [comment]
>We decided to manage the scheduled posts in the same way as the published posts, that is using the remote auto save to create an unpublished revision to be shown as a remote preview to the user without affecting the published (or scheduled) version.

I was simply duplicating the behavior from Calypso when I implemented the changes in `master-remote-auto-save`. Calypso treats `Scheduled` posts as `Drafts`. 
I'm not sure if I wasn't part of that discussion or if I just forgot. However, we need to make sure the autosave endpoint works as expected for `Scheduled` posts. If it does, we can simply modify [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/48d95ebeaa419601448b39404db732b4ca0663b8/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java#L243) if statement and everything else should work out of the box.

**Comment by develric** [comment]
Hi @malinajirka , thanks for comment on this, really appreciated 😊.

> I'm not sure if I wasn't part of that discussion or if I just forgot

You are perfectly right, this was a shared idea on a slack chat with @maxme ; as general approach I actually listed some open items emerged while working on [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10085) and started a dedicated ticket for each to further discuss. This ticket is part of those open items but I have to admit that reading again what I wrote I treated this one more as a decision than a discussion and my bad for this! I will put more attention 🙇‍♂️. So please take this as an open discussion, thanks. (also don't know if it makes sense that other need to be involved, in case feel free to cc of course).

> Calypso treats Scheduled posts as Drafts.

One point that was driving in favor to the management of scheduled posts as published (and so to use remote auto save while modifying for previewing) was the fact that the scheduled date can be close to when you are editing, so at least AFAIU we should put some mechanism in place to avoid you end up publishing a content you are just modifying on the fly for preview (and remote auto save could be one possibility in that direction). But I have to admit I have not final ideas on this. And also the fact that Calypso manages scheduled as draft would guide to align in that direction.
Hi @maxme, any advise?

PS: maybe it makes sense to move this discussion in a dedicated ticket in WP-Android? wdyt?

**Comment by develric** [comment]
Hi @maxme and @malinajirka 😊. 
As agreed I made some checks on the WP.com API with a rest client, in order to be more confident that remote auto save works as expected with scheduled posts before to decide if to go with the FluxC modifications. 

### Tl;Dr:
AFAIU the remote auto save in WP.com API is working fine also for Scheduled posts.

### Some background
To better compare I made the same checks with a Published post and with a Scheduled post. I tried to be as detailed as I think it is sufficient so that someone else could reproduce eventually (probably obvious for many, hopefully useful for someone 😊). On the web site I checked both from block editor and from classic editor (not fully sure if it makes any difference but behavior seems aligned at least for the checks done).

### Tools used
I used ARC as a rest client (but everything else capable of OAuth2 authentication should fit)

### Prerequisites
- WP.com site: you need to have a WordPress account with a WP.com site to test and make checks on the web side
- OAuth2 authentication configured: [here](https://github.com/wordpress-mobile/WordPress-Android#oauth2-authentication) some more details
- SITE_ID and POST_ID: given you are in the editor editing a post you can get them from the URL as follows `https://wordpress.com/block-editor/post/SITE_ID/POST_ID`
- Create (if not yet available) a Published and a Scheduled post for the test

### TEST CASE 1 - POST REMOTE AUTO SAVE ON A PUBLISHED POST 
**Final Result: passed** ✔
- [ ] Open a published post in the web editor and take note of the text content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60402895-07d04480-9b96-11e9-9460-1093e5530fad.png">
</kbd>
</p>

- [ ] with ARC send a POST to 
`https://public-api.wordpress.com/rest/v1.1/sites/SITE_ID/posts/POST_ID/autosave/`
configuring the body for `application/json` and the following content

``` JSON
{
  "title": "published post",
  "excerpt": "",
  "content": "remote auto saved by ARC rest client"
}
```
- [ ] Check that on the web the editor signal there is an autosave and you get the expected content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403033-030c9000-9b98-11e9-917e-10d6f14cd985.png">
</kbd>
</p>
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403041-2e8f7a80-9b98-11e9-9332-0e7d85b37b86.png">
</kbd>
</p>
- [ ] Check in ARC you get response code 200 (OK) and a content similar to the following

``` JSON
{
"ID": 371,
"post_ID": 369,
"modified": "2019-06-30T22:34:16+00:00",
"preview_URL": "http://SITE_ID/2019/06/30/published-post/?preview=true&preview_id=371&preview_nonce=47a0485869"
}
```
- [ ] Open the got preview_URL link in a browser and you should get the correct autosaved post content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403102-54694f00-9b99-11e9-8715-9f63a3b4d92b.png">
</kbd>
</p>

### TEST CASE 2 - GET REMOTE AUTO SAVE ON A PUBLISHED POST 
**Final Result: passed** ✔
- [ ] with ARC send a GET to 
`https://public-api.wordpress.com/rest/v1.1/sites/SITE_ID/posts/POST_ID/autosave/`
- [ ] Check in ARC you get response code 200 (OK) and a content similar to the following

``` JSON
{
"ID": 371,
"author_ID": "153135329",
"post_ID": 369,
"title": "published post",
"content": "remote auto saved by ARC rest client",
"excerpt": "",
"preview_URL": "http://SITE_ID/2019/06/30/published-post/?preview=true&preview_nonce=47a0485869",
"modified": "2019-06-30T22:34:16+00:00"
}
```
- [ ] Open the got preview_URL link in a browser and you should get the correct autosaved post content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403102-54694f00-9b99-11e9-8715-9f63a3b4d92b.png">
</kbd>
</p>

### TEST CASE 3 - POST REMOTE AUTO SAVE ON A SCHEDULED POST 
**Final Result: passed** ✔
- [ ] Open a scheduled post in the web editor and take note of the text content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403169-3d772c80-9b9a-11e9-9c64-6e16f7ad4eab.png">
</kbd>
</p>

- [ ] with ARC send a POST to 
`https://public-api.wordpress.com/rest/v1.1/sites/SITE_ID/posts/POST_ID/autosave/`
configuring the body for `application/json` and the following content

``` JSON
{
  "title": "scheduled post",
  "excerpt": "",
  "content": "remote auto saved by ARC rest client"
}
```
- [ ] Check that on the web the editor signal there is an autosave and you get the expected content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403194-78796000-9b9a-11e9-9ab0-1205a3e0b376.png">
</kbd>
</p>
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403205-88913f80-9b9a-11e9-8221-335559434bc2.png">
</kbd>
</p>

- [ ] Check in ARC you get response code 200 (OK) and a content similar to the following

``` JSON
{
"ID": 367,
"post_ID": 343,
"modified": "2019-06-30T22:40:35+00:00",
"preview_URL": "http://SITE_ID/?p=343&preview=true&preview_id=367&preview_nonce=152f6db8a4"
}
```
- [ ] Open the got preview_URL link in a browser and you should get the correct autosaved post content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403223-d1e18f00-9b9a-11e9-9d0f-ad6801393f14.png">
</kbd>
</p>

### TEST CASE 4 - GET REMOTE AUTO SAVE ON A SCHEDULED POST 
**Final Result: passed** ✔
- [ ] with ARC send a GET to 
`https://public-api.wordpress.com/rest/v1.1/sites/SITE_ID/posts/POST_ID/autosave/`
- [ ] Check in ARC you get response code 200 (OK) and a content similar to the following

``` JSON
{
"ID": 367,
"author_ID": "153187329",
"post_ID": 343,
"title": "scheduled post",
"content": "remote auto saved by ARC rest client",
"excerpt": "",
"preview_URL": "http://SITE_ID/?p=343&preview=true&preview_nonce=152f6db8a4",
"modified": "2019-06-30T22:40:35+00:00"
}
```
- [ ] Open the got preview_URL link in a browser and you should get the correct autosaved post content
<p align="center">
<kbd>
  <img width="300" src="https://user-images.githubusercontent.com/47797566/60403102-54694f00-9b99-11e9-8715-9f63a3b4d92b.png">
</kbd>
</p>

**Comment by malinajirka** [comment]
Thank you @develric for the thorough tests! ;) I'll open a PR in a bit.
